dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -633,6 +633,25 @@ private String findConfigedHosts(ProtocolConfig protocolConfig, List<URL> regist
      * @return
      */
     private Integer findConfigedPorts(ProtocolConfig protocolConfig, String name, Map<String, String> map) {
+        Integer portToBind = getPortFromProtocol(protocolConfig, name);
+
+        // save bind port, used as url's key later
+        map.put(Constants.BIND_PORT_KEY, String.valueOf(portToBind));

```

Comment:
method name indicate a search operation based on method inputs, should we modify input object here?